modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java

```diff
@@ -1138,14 +1138,15 @@ public void processJobExecuteResponse(UUID nodeId, GridJobExecuteResponse msg) {
         lock.readLock();
 
         try {
+            GridTaskWorker<?, ?> task = tasks.get(msg.getSessionId());
+
             if (stopping && !waiting) {
-                U.warn(log, "Received job execution response while stopping grid (will ignore): " + msg);
+                U.warn(log, "Received job execution response while stopping grid (will ignore): " + msg
+                    + (task != null && task.getSession() != null ? (", " + task.getSession().getTaskName()) : ""));
```

Comment:
The similar code block repeats 5 times. It should be refactored to separate method, like tryResolveTaskName(sesId)

Reply:
Good idea.
However, I think it will be more clear to use task as parameter. Otherwise, here will be one more place which works with tasks map.